src/lib/metrics.py

```diff
@@ -43,21 +47,55 @@ class IngestLine:
     dimension_values: List[DimensionValue]
 
     def dimensions_string(self) -> str:
-        dimension_values = [f'{dimension_value.name[0:ALLOWED_METRIC_DIMENSION_KEY_LENGTH]}="{dimension_value.value[0:ALLOWED_METRIC_DIMENSION_VALUE_LENGTH]}"'
-                            for dimension_value
-                            in self.dimension_values
-                            if dimension_value.value != ""]  # MINT rejects line with empty dimension value
+        dimension_values = [
+            f'{dimension_value.name[0:ALLOWED_METRIC_DIMENSION_KEY_LENGTH]}="{dimension_value.value[0:ALLOWED_METRIC_DIMENSION_VALUE_LENGTH]}"'
+            for dimension_value in self.dimension_values
+            if dimension_value.value != ""
+        ]  # MINT rejects line with empty dimension value
         dimensions = ",".join(dimension_values)
         if dimensions:
             dimensions = "," + dimensions
         return dimensions
 
     def to_string(self) -> str:
-        separator = ',' if self.metric_type == 'gauge' else '='
-        metric_type = self.metric_type if self.metric_type != 'count' else 'count,delta'
+        separator = "," if self.metric_type == "gauge" else "="
+        metric_type = self.metric_type if self.metric_type != "count" else "count,delta"
         return f"{self.metric_name[0:ALLOWED_METRIC_KEY_LENGTH]}{self.dimensions_string()} {metric_type}{separator}{self.value} {self.timestamp}"
 
 
+@dataclass(frozen=True)
+class MetadataIngestLine(IngestLine):
+    metric_name: str
+    metric_type: str
+    meta_metric_display_name: str
+    meta_metric_description: str
+    meta_metric_unit: str
+
+    def __init__(self, **kwargs):
+        object.__setattr__(self, "metric_name", kwargs.get("metric_name", ""))
+        object.__setattr__(
+            self,
+            "metric_type",
+            "count" if "count" in kwargs.get("metric_type", "") else kwargs.get("metric_type", ""),
+        )
+        object.__setattr__(
+            self,
+            "meta_metric_display_name",
+            f"[Autodiscovered] {kwargs.get('metric_display_name', '')}"
+            if len(kwargs.get("metric_display_name", "")) > 0
+            else f"[Autodiscovered] {kwargs.get('metric_name', '')}",
+        )
+        object.__setattr__(self, "meta_metric_description", kwargs.get("metric_description", ""))
+        object.__setattr__(self, "meta_metric_unit", kwargs.get("metric_unit", ""))
+
+    def to_string(self) -> str:
+        return (f'#{self.metric_name[0:ALLOWED_METRIC_KEY_LENGTH]} {self.metric_type} '
```

Comment:
you can use ''' for multiline strings

Reply:
I cant use them with f-strings
https://stackoverflow.com/questions/70863543/can-a-python-docstring-be-calculated-f-string-or-expression/70865657#70865657

Reply:
good

Reply:
you can use `\` to break line